components/server/src/ome/services/util/Executor.java

```diff
@@ -354,6 +364,11 @@ public Impl(CurrentDetails principalHolder, SessionFactory factory,
             this.service = service;
             // Allowed to create more threads.
             this.systemService = Executors.newCachedThreadPool();
+            this.backgroundService = Executors.newFixedThreadPool(10,
```

Comment:
Would be great if this were configurable (so we can experiment) and maybe wise to default to less than default DB pool size.

Reply:
Agreed. This is very much a temporary fix for testing purposes while we figure out the appropriate concurrency abstraction.